src/hint/Substitutions.cpp

```diff
@@ -49,6 +43,16 @@ void tvm::hint::internal::Substitutions::add(const Substitution & s)
   }
 }
 
+void Substitutions::remove(const Substitution & s)
+{
+  auto it = std::find_if(substitutions_.begin(), substitutions_.end(), [&](const auto & e) { return &e == &s; });
+  if(it == substitutions_.end())
+    throw std::runtime_error("[Substitutions::remove] Substitution s is not part of the substitutions.");
```

Comment:
Since we silently ignore removing tasks that are not present we should do the same with substitutions (e.g. one might remove a substitution that is already removed by removing the associated task)

Reply:
Changed in 5366d455dfca17ce396d1648f583e890e457600c